robot/basestation/static/js/elements/navigation/localization.js

```diff
@@ -0,0 +1,129 @@
+let rover_heading = 0
+let rover_x = 0
+let rover_y = 0
+let zoom_value = 1
+let MAX_ZOOM = 2
+let MIN_ZOOM = 0.25
+let canvas_pos
+let keys = { //javascript keycodes
+    'r': 82,
+    'w': 87,
+    'a': 65,
+    's': 83,
+    'd': 68
+}
+
+let base_lat
+let base_long
+$(document).ready(() => {
+    // the elements might not be initialized yet, must test
+    base_lat = $('#antenna-latitude').text()
+    base_long = $('#antenna-longitude').text()
+})
+
+function setup() {
+    let canvas = createCanvas(450, 300)
+    canvas.parent('#canvasContainer')
+    canvas_pos = $('#defaultCanvas0').offset()
+    angleMode(DEGREES)
+}
+
+function draw() {
+    // all of the manual controls
+    if (keyIsDown(keys['r'])) { //reset position
+        rover_heading = 0;
+        zoom_value = 1;
+        rover_x = rover_y = 0;
+    }
+    if (keyIsDown(keys['a'])) { // spin ccw
+        rover_heading -= 1
+    }
+    if (keyIsDown(keys['d'])) { // spin cw
+        rover_heading += 1
+    }
+    adjustmap_rotation_deg() // keep rover_heading within 0-360
+    if (keyIsDown(keys['w'])) { // move forward
+        rover_x += sin(rover_heading);
+        rover_y -= cos(rover_heading);
+    }
+    if (keyIsDown(keys['s'])) { // move backward
+        rover_x -= sin(rover_heading);
+        rover_y += cos(rover_heading);
+    }
+
+    // canvas static
+    background(225) // draw background
+    translate(width / 2, height / 2) // make the (0,0) point the center of canvas
+    scale(zoom_value) // zoom to given value
+    draw_grid() // draw the grid for reference
+    circle(0, 0, 10) // draw circle base station
```

Comment:
why 10? magic number

Reply:
is this circle always in the same location? does it move? if there's an initial position you might want to define an array called basestation_initial_pos or something with the coordinates / values here, just throwing an idea out there

Reply:
this draws a circle at point 0,0 with diameter of 10px